src/Cake.Frosting/FrostingTask.cs

```diff
@@ -34,6 +35,16 @@ public virtual void Run(T context)
         }
 
         /// <summary>
+        /// Runs the task using the specified context.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        public virtual Task RunAsync(T context)
```

Comment:
Perhaps we should add a `AsyncFrostingTask` instead of having two Run methods which might be confusing. One could (wrongly) assume that both methods are executed. Having a single `Run` method makes the API cleaner as well.

For the standard task type we could explicitly implement the interface.

```csharp
public class FrostingTask : IFrostingTask
{
  void IFrostingTask.RunAsync(T context) 
  {
    Run(context);
    return Task.CompletedTask;
  }

  public virtual void Run(T context)
  {
  }
}
```

And the async `FrostingTask` could be something like this:

```csharp
public class AsyncFrostingTask : IFrostingTask
{
  public virtual Task Run(T context)
  {
  }
}
```